pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2867,8 +2867,10 @@ func fetchEnvVariablesFromCloudInit(config types.DomainConfig) (map[string]strin
 	envList := make(map[string]string, 0)
 	list := strings.Split(string(ud), "\n")
 	for _, v := range list {
-		pair := strings.SplitN(v, ":", 2)
-		envList[pair[0]] = pair[1]
+		pair := strings.SplitN(v, "=", 2)
+		if len(pair) == 2 {
+			envList[pair[0]] = pair[1]
+		}
```

Comment:
How about returning an if you have less than 2 parts?

Hmm, if these are shell env it is perfectly valid to set an env to nothing e.g,
  DEBUG= myCommand
or
 export DEBUG=

Basically the value is empty, but the variable exists in the environment.

Reply:
Hmm. Am wondering how to flag the Error. We should not silently ignore errors.. We need to provide a way to make them visible to the user. I see 2 choices:

1) Return the error string to the user - but go ahead with AI creation
2) Fail the AI creation with Cloud-init parsing error - with details of the error.

I prefer (2) - Keeps things simple.

Reply:
Some Error that says:

Parsing Error for Cloud Init - Variable <FOO> not defined correctly. Variables should be defined in "<Var>=<val>" OR "<Var>="  ( Empty Variable ).